lib/src/summarize.dart

```diff
@@ -4,12 +4,51 @@
 
 library dart_pad.summarize;
 
+import '../services/dartservices.dart';
 /// Instances of this class take string input of dart code and output a text 
 /// description ofthe code's size, packages, and other useful information
 class Summarizer {
   
-  String summarize(String input) {
-    return "foo";
+  String summarize(SourceRequest input, AnalysisResults result) {
+    if (input == null || result == null) {
```

Comment:
`input` should never be null - it's a programmer error if is is. We can use as `assert`, or do something like, `if (input == null) throw new ArgumentError('input can't be null')`, _or_ just just wait for an NPE later on in the method. This method depends so heavily on the source code, that throwing an NPE if it's not present is ok.